docs/advanced/tls_porting.md

```diff
@@ -14,61 +14,77 @@ Creating session keys is only one use for random values; they have far more comp
+mbed TLS distinguishes between strong and weak entropy sources. Of the sources registered by default, two are strong: /dev/urandom and Windows CryptoAPI. However, these resources are not available on many embedded platforms, and the default behaviour of mbed TLS is to refuse to work if there are no strong sources present. To get around this, mbed TLS assumes that the hardware entropy source you register (as explained in this section) is a TRNG and thus treats it as strong.
+
+It is very important that you only add a TRNG the way described in this section. For the purposes of this document a device is considered a TRNG only if it is dedicated to generating entropy to be used in cryptographic applications, and careful consideration has been given to how much the data generated is subject to adversarial manipulation and to how much entropy it can actually provide.
+
+For example, an integrated circuit extracting statistically random data from two oscillators of unknown frequencies and independent phases is considered as a TRNG, while anything derived from a real time clock is NOT.
 
 The preferred way to provide a custom entropy source:
 
-1. Implement the `mbedtls_hardware_poll` function to let it access the device's entropy source.
-2. Set the appropriate macros for your target in `hal/targets.json`.
+1. Provide an implementation of the functions defined in `trng_api.h` to let mbed TLS access the device's entropy source.
+2. Indicate that your target has an entropy source in `hal/targets.json`, by adding `TRNG` to your device's `device_has` section.
 
 The next two sections explain how to do this.
 
-## How to implement the hardware poll
+## How to implement the TRNG API
+
+The implementation of this interface goes to the mbed OS directory specific to your target. (The name of this directory is of the form `TARGET_<target name>`, for example in the case of K64F targets it is `TARGET_MCU_K64F`)
+
+### Data structure
 
-The `mbedtls_hardware_poll` function is declared in `entropy_poll.h`:
+You have to define a structure `trng_s` that holds all the information needed to operate the peripheral and describe its state.
+
+### Initialisation and deinitialisation
+
+You need to provide means for initialising and deinitialising the peripheral with the following functions:
```

Comment:
Style: I am not sure about the word `deinitialise`. I would suggest `release`, `free`, etc.